crates/hir/src/lib.rs

```diff
@@ -488,6 +488,20 @@ impl Module {
         Some(Module { id: def_map.module_id(parent_id) })
     }
 
+    /// Finds nearest non-block ancestor `Module` (`self` included).
+    fn nearest_non_block_module(self, db: &dyn HirDatabase) -> Module {
+        let mut id = self.id;
+        loop {
+            let def_map = id.def_map(db.upcast());
+            let origin = def_map[id.local_id].origin;
+            if matches!(origin, ModuleOrigin::BlockExpr { .. }) {
```

Comment:
Is there a simpler way to check whether a `Module` is a block? I thought I could do better but failed badly...

Reply:
the `ModuleId` contained in a `hir::Module` has a `block` field that's set when its a block

Reply:
Ah right, then I think this is the appropriate way